src/core/server/http/http_service.mock.ts

```diff
@@ -30,6 +32,7 @@ const createSetupContractMock = () => {
     setBasePathFor: jest.fn(),
     // we can mock some hapi server method when we need it
     server: {} as Server,
+    createNewServer: async (cfg: HttpConfig): Promise<HttpServerSetup> => ({} as HttpServerSetup),
```

Comment:
This arrow function doesn't need to be marked as async.

Reply:
When I remove `async` I get:

```
{
	"resource": "/Users/todd/src/kibana/src/core/server/http/http_service.mock.ts",
	"owner": "typescript",
	"code": "2739",
	"severity": 8,
	"message": "Type 'HttpServerSetup' is missing the following properties from type 'Promise<HttpServerSetup>': then, catch, [Symbol.toStringTag], finally",
	"source": "ts",
	"startLineNumber": 35,
	"startColumn": 69,
	"endLineNumber": 35,
	"endColumn": 92
}
```

Reply:
I could make it return a promise that resolves the empty cast object, but async seemed to be a faster way to do that :)